parse.go

```diff
@@ -17,37 +19,76 @@ import (
 //
 //  /_-
 type Parser struct {
+	ParserConfig
 }
 
+// ParserConfig is a set of configuration options for the Parser.
+type ParserConfig struct {
+	// Variant is the tags sytax to parse for.
+	Variant HashtagVariant
+}
+
+// HashtagVariant represents one of the different flavours of hashtag syntax.
+type HashtagVariant uint
+
+const (
+	DefaultVariant  HashtagVariant = 0
+	ObsidianVariant                = iota
```

Comment:
Also, instead of `0` and `iota`, perhaps we should do just `iota`:

```
const (
  DefaultVariant Variant = iota
  ObsidianVariant
)
```

The second one will pick up the type and "= iota" automatically.